Source/LinuxParking.API/Controllers/SpotController.cs

```diff
@@ -1,49 +1,108 @@
-        public SpotController()
+        private readonly ISpotService _spotService;
+        private readonly IMapper _mapper;
+
+        public SpotController(ISpotService spotService, IMapper mapper)
         {
-            // TODO: Dependecy injections
+            _spotService = spotService;
+            _mapper = mapper;
         }
 
         [HttpPost]
-        public async Task CreateAsync([FromRoute] string stationId)
+        public async Task<IActionResult> CreateAsync([FromRoute] int stationId, [FromBody] CreateSpotResource resource)
         {
-            throw new NotImplementedException();
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState.GetErrorMessages());
+            }
+
+            var spot = _mapper.Map<CreateSpotResource, Spot>(resource);
+            spot.StationId = stationId;
+            var res = await _spotService.SaveAsync(spot);
+
+            if (!res.Success)
+            {
+                return BadRequest(res.Message);
+            }
+
+            var spotResource = _mapper.Map<Spot, SpotResource>(res.Spot);
+            return Created("", spotResource);
         }
 
         [HttpGet]
-        public async Task GetAllAsync([FromRoute] string stationId)
+        public async Task<IActionResult> GetAllAsync([FromRoute] int stationId)
         {
-            throw new NotImplementedException();
+            var res = await _spotService.ListAsync(stationId);
+
+            if (!res.Success)
+            {
+                return BadRequest(res.Message);
+            }
+
+            var spotResponse = _mapper.Map<IEnumerable<Spot>, IEnumerable<SpotResource>>(res.Spots);
+            return Ok(spotResponse);
         }
 
         [HttpGet]
         [Route("{spotId}")]
-        public async Task GetAsync([FromRoute] string stationId, [FromRoute] string spotId)
+        public async Task<IActionResult> GetAsync([FromRoute] int stationId, [FromRoute] int spotId)
         {
-            throw new NotImplementedException();
+            var res = await _spotService.FindByIdAsync(stationId, spotId);
+            if (!res.Success)
+            {
+                return BadRequest(res.Message);
+            }
+
+            var spotResponse = _mapper.Map<Spot, SpotResource>(res.Spot);
+            return Ok(spotResponse);
         }
 
         [HttpPut]
         [Route("{spotId}")]
-        public async Task UpdateAsync([FromRoute] string stationId, [FromRoute] string spotId)
+        public async Task<IActionResult> UpdateAsync(int stationId, int spotId, [FromBody] CreateSpotResource resource)
         {
-            throw new NotImplementedException();
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState.GetErrorMessages());
+            }
+
+            var spot = _mapper.Map<CreateSpotResource, Spot>(resource);
+            var res = await _spotService.UpdateAsync(stationId, spotId, spot);
+
+            if (!res.Success)
```

Comment:
We should follow the same "style" trough the whole project.
If you could change this to:
```csharp
if (stmt)
    return <value>
```